contracts/src/account/mod.rs

```diff
@@ -1,27 +1,16 @@
-use near_sdk::{ext_contract, near, AccountId};
+use near_contract_standards::non_fungible_token::{
+    core::NonFungibleTokenCore, NonFungibleTokenEnumeration, NonFungibleTokenResolver,
+};
+use near_sdk::{ext_contract, AccountId};
 
-pub use self::error::AccountError;
-
-mod error;
-
-// TODO: make this contract an extension of NFT, since every Account
-// is a unique item. This will ease the integration of Defuse
-// with already existing tooling around NFTs
 #[ext_contract(ext_account_contract)]
-pub trait AccountContract {
+pub trait AccountContract:
+    NonFungibleTokenCore + NonFungibleTokenResolver + NonFungibleTokenEnumeration
+{
     /// Create an account with given defivation path for given owner
-    fn create_account(&mut self, owner: AccountId, derivation_path: String);
-    /// Change an owner of account with given derivation path.
-    fn change_owner(&mut self, from: &AccountId, to: AccountId, derivation_path: String);
-    /// Return all [`Account`]s owned by given `owner`
-    fn get_accounts(&self, owner: &AccountId) -> Vec<(String, Account)>;
-
-    /// Return MPC contract for this
-    fn mpc_contract(&self) -> &AccountId;
-}
-
-#[derive(Debug, Clone, Default, Eq, PartialEq)]
-#[near(serializers=[borsh, json])]
-pub struct Account {
-    is_locked: bool,
+    /// By default, owner is sender
+    // TODO: maybe accept optional derivation path, so it can be also generated on-chain?
+    // TODO: derivation_path can contain owner_id, so it can be generated
+    // off-chain and registration can not be front-runned
+    fn create_account(&mut self, derivation_path: String, owner: Option<AccountId>);
```

Comment:
I didn't pay too much attention to this logic since we haven't decided yet about the format of derivation path and how exactly it should work, but the core idea here is to be able to "claim" derivation path in the name of `env::predecessor_id()` by default. Or to let anyone send a tx and pay gas to claim for other user.